portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,13 +36,20 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Comment").addSort("date", SortDirection.DESCENDING);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Comment").addSort("date", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
 
+    String numberOfCommentsString = request.getParameter("number-comments");
+    // Set limit on number of comments in results (Default: Display all comments)
+    int limit = numberOfCommentsString == null || numberOfCommentsString.isEmpty() ? Integer.MAX_VALUE : Integer.parseInt(numberOfCommentsString);
```

Comment:
Nit: Just because the ternary expression here is a little lengthy, I'll use this as an example of what we spoke about offline, where you could potentially extract some part of this out into a helper method, whether that's just the conditional check of whether there is a number so the reader doesn't have to parse between two things, or, to make it even less verbose, to get the number of comments as a number type (and you could return undefined if empty) so here it could look something like
` int limit = getNumberComments(numberCommentsString) || Integer.MAX_VALUE`

Reply:
That's a good point, even though it gets the job done, it's pretty ugly to look at and not as readable as could be with a helper function. I also appreciate you pointing out an example of what we've talked about, that's really helpful :).